submiting homework

nizamimursudlu opened the pull request.

eidosam commented on Week1/homework/js-exercises/ex1-bookList.js:
```diff
@@ -0,0 +1,66 @@
+secondBook.style.cssText = "background-color:red; padding-left: 20px; width: 400px"
+thirdBook.style.cssText = "background-color:green; padding-left: 20px; width: 400px"
+
+document.querySelector("#bookList").style.cssText = "display: flex; flex-direction: row; justify-content: space-evenly;"
```
Please re-write this exercise with using `for` loop, or `forEach`, or any other one you are familiar with. Will re-visit this exercise once you do so.

nizamimursudlu replied:
Hello Samir,
Thank you for your feedback!
I did all the corrections you mentioned, could you please have a look at my homework again?
Thanks a lot!

eidosam reviewed (changes requested):
Good work in general, Nizami! I left some inline notes and requests, please have a look.
Besides them, there are some general notes:
1. Try to use `const` whenever you declare a variable that won't be re-assigned.
2. Please use [gitignore](https://www.atlassian.com/git/tutorials/saving-changes/gitignore) to exclude files are irrelative to the code you are writing, like .DS_STORE and .vscode/
3. Try to stick to the file structure you were provided, so use the files in `Week1/js-exercises` and `Week1/project` instead of creating new directories.

eidosam commented on Week1/homework/js-exercises/ex1-bookList.js:
```diff
@@ -0,0 +1,43 @@
+  image.src = arrImages[i]
+  const paragraph = document.createElement("p")
+  paragraph.innerHTML = `${books[i].title} by ${books[i].author} `
+  book.appendChild(createBookList(books[i]).appendChild(paragraph)).appendChild(image)
```
This chaining need to be reviewed, it doesn't do what expected, you need to re-arrange the parenthesis to fix it.
Specifically this

    book.appendChild(createBookList(books[i]).appendChild(paragraph))

This will append only the paragraph to the `bookList` div. Because 

    createBookList(books[i]).appendChild(paragraph)

Returns only the appended child.

eidosam replied:
The function `createBookList` you created above is designed to receive the content as text, but here you are passing the whole book object.

eidosam replied:
The request in this exercise is to
> Use a \<ul\> and \<li\> to display the books.

But this is how the structure looks like
<img width="858" alt="nizami-ex1" src="https://user-images.githubusercontent.com/17527017/102938098-ea8ae680-44ab-11eb-9f0f-4114db04c417.png">

Can you please fix this?

eidosam commented on Week1/homework/js-exercises/ex2-aboutMe.js:
```diff
@@ -0,0 +1,17 @@
+}
+document.getElementById("nickname").innerHTML = `${"Nizami Mursudlu"}`
+document.getElementById("fav-food").innerHTML = `${"Pizza"}`
+document.getElementById("hometown").innerHTML = `${"Baku"}`
```
The template-literal pattern is used when you want to compose a text from combination of text and variables, but here you can simply use regular strings

    document.getElementById("nickname").innerHTML = "Nizami Mursudlu"

eidosam reviewed (changes requested):
There is good progress in this PR! Thank you!
I added new inline comments about exercise 1, please have a look!

Another thing; can you please remove the files which are not related to this homework? Or the duplicate files. There are a lot of conflicts in the files in this directory `Week1/js-exercises/`.

nizamimursudlu commented:
Hello Samir, hope you had a good xmas
Thank you for your feedback.

I got a little confused by your last screenshot… Why do I have to write JS if everything is already displayed in HTML…

I changed code. Please have a look…



> On 22 Dec 2020, at 23:26, Samir Aleido <notifications@github.com> wrote:
> 
> 
> @eidosam requested changes on this pull request.
> 
> There is good progress in this PR! Thank you!
> I added new inline comments about exercise 1, please have a look!
> 
> Another thing; can you please remove the files which are not related to this homework? Or the duplicate files. There are a lot of conflicts in the files in this directory Week1/js-exercises/.
> 
> In Week1/homework/js-exercises/ex1-bookList.js <https://github.com/HackYourHomework/JavaScript2/pull/11#discussion_r547514131>:
> 
> > +  author: 'Antoine De Saint-Exupéry',
> +  alreadyRead: true
> +},
> +{
> +  title: 'Veronica Decides To Die',
> +  author: 'Paulo Coelho',
> +  alreadyRead: false
> +},
> +{
> +  title: 'Anna Karenina',
> +  author: 'Leo Tolstoy',
> +  alreadyRead: true
> +}
> +];
> +const book = document.querySelector("#bookList")
> +const ulElement = createBookList(books);
> This element is not used anywhere, maybe you can remove it.
> 
> In Week1/homework/js-exercises/ex1-bookList.js <https://github.com/HackYourHomework/JavaScript2/pull/11#discussion_r547515321>:
> 
> > +  title: 'Anna Karenina',
> +  author: 'Leo Tolstoy',
> +  alreadyRead: true
> +}
> +];
> +const book = document.querySelector("#bookList")
> +const ulElement = createBookList(books);
> +const arrImages = ["https://www.booktopia.com.au/http_coversbooktopiacomau/big/9781853261589/0000/the-little-prince.jpg",
> +  "https://images-na.ssl-images-amazon.com/images/I/411YxgRv1bL._SX324_BO1,204,203,200_.jpg",
> +  "https://images-na.ssl-images-amazon.com/images/I/51vPf2CfSEL.jpg"]
> +
> +function createBookList(books) {
> +  let li = document.createElement('li');
> +  li.textContent = books;
> +  return li
> +}
> The functions seems to create a single item, not list, so maybe you can rename it to reflect this? Also the argument should be singular (book).
> 
> In Week1/homework/js-exercises/ex1-bookList.js <https://github.com/HackYourHomework/JavaScript2/pull/11#discussion_r547515852>:
> 
> > +  alreadyRead: true
> +}
> +];
> +const book = document.querySelector("#bookList")
> +const ulElement = createBookList(books);
> +const arrImages = ["https://www.booktopia.com.au/http_coversbooktopiacomau/big/9781853261589/0000/the-little-prince.jpg",
> +  "https://images-na.ssl-images-amazon.com/images/I/411YxgRv1bL._SX324_BO1,204,203,200_.jpg",
> +  "https://images-na.ssl-images-amazon.com/images/I/51vPf2CfSEL.jpg"]
> +
> +function createBookList(books) {
> +  let li = document.createElement('li');
> +  li.textContent = books;
> +  return li
> +}
> +
> +for (i = 0; i < 3; i++) {
> Instead of hard-coding the number of elements (3), you can use books.length
> 
> In Week1/homework/js-exercises/ex1-bookList.js <https://github.com/HackYourHomework/JavaScript2/pull/11#discussion_r547532018>:
> 
> > +const arrImages = ["https://www.booktopia.com.au/http_coversbooktopiacomau/big/9781853261589/0000/the-little-prince.jpg",
> +  "https://images-na.ssl-images-amazon.com/images/I/411YxgRv1bL._SX324_BO1,204,203,200_.jpg",
> +  "https://images-na.ssl-images-amazon.com/images/I/51vPf2CfSEL.jpg"]
> +
> +function createBookList(books) {
> +  let li = document.createElement('li');
> +  li.textContent = books;
> +  return li
> +}
> +
> +for (i = 0; i < 3; i++) {
> +  const image = document.createElement("img");
> +  image.src = arrImages[i]
> +  const paragraph = document.createElement("p")
> +  paragraph.innerHTML = `${books[i].title} by ${books[i].author} `
> +  book.appendChild(createBookList(books[i]).appendChild(paragraph)).appendChild(image)
> This chaining need to be reviewed, it doesn't do what expected, you need to re-arrange the parenthesis to fix it.
> Specifically this
> 
> book.appendChild(createBookList(books[i]).appendChild(paragraph))
> This will append only the paragraph to the bookList div. Because
> 
> createBookList(books[i]).appendChild(paragraph)
> Returns only the appended child.
> 
> In Week1/homework/js-exercises/ex1-bookList.js <https://github.com/HackYourHomework/JavaScript2/pull/11#discussion_r547532618>:
> 
> > +const arrImages = ["https://www.booktopia.com.au/http_coversbooktopiacomau/big/9781853261589/0000/the-little-prince.jpg",
> +  "https://images-na.ssl-images-amazon.com/images/I/411YxgRv1bL._SX324_BO1,204,203,200_.jpg",
> +  "https://images-na.ssl-images-amazon.com/images/I/51vPf2CfSEL.jpg"]
> +
> +function createBookList(books) {
> +  let li = document.createElement('li');
> +  li.textContent = books;
> +  return li
> +}
> +
> +for (i = 0; i < 3; i++) {
> +  const image = document.createElement("img");
> +  image.src = arrImages[i]
> +  const paragraph = document.createElement("p")
> +  paragraph.innerHTML = `${books[i].title} by ${books[i].author} `
> +  book.appendChild(createBookList(books[i]).appendChild(paragraph)).appendChild(image)
> The function createBookList you created above is designed to receive the content as text, but here you are passing the whole book object.
> 
> In Week1/homework/js-exercises/ex1-bookList.js <https://github.com/HackYourHomework/JavaScript2/pull/11#discussion_r547534238>:
> 
> > +const arrImages = ["https://www.booktopia.com.au/http_coversbooktopiacomau/big/9781853261589/0000/the-little-prince.jpg",
> +  "https://images-na.ssl-images-amazon.com/images/I/411YxgRv1bL._SX324_BO1,204,203,200_.jpg",
> +  "https://images-na.ssl-images-amazon.com/images/I/51vPf2CfSEL.jpg"]
> +
> +function createBookList(books) {
> +  let li = document.createElement('li');
> +  li.textContent = books;
> +  return li
> +}
> +
> +for (i = 0; i < 3; i++) {
> +  const image = document.createElement("img");
> +  image.src = arrImages[i]
> +  const paragraph = document.createElement("p")
> +  paragraph.innerHTML = `${books[i].title} by ${books[i].author} `
> +  book.appendChild(createBookList(books[i]).appendChild(paragraph)).appendChild(image)
> The request in this exercise is to
> 
> Use a <ul> and <li> to display the books.
> 
> But this is how the structure looks like
>  <https://user-images.githubusercontent.com/17527017/102938098-ea8ae680-44ab-11eb-9f0f-4114db04c417.png>
> Can you please fix this?
> 
> In Week1/homework/js-exercises/ex2-aboutMe.js <https://github.com/HackYourHomework/JavaScript2/pull/11#discussion_r547535196>:
> 
> > +
> +
> +const myImage = document.createElement("img");
> +myImage.src = "decent-pic.jpeg";
> +document.body.appendChild(myImage);
> +myImage.style.width = "200px"
> +
> +
> +const lis = document.querySelectorAll("ul > li")
> +for (let i = 0; i < 3; i++) {
> +  const li = lis[i];
> +  li.classList.add("list-item");
> +}
> +document.getElementById("nickname").innerHTML = `${"Nizami Mursudlu"}`
> +document.getElementById("fav-food").innerHTML = `${"Pizza"}`
> +document.getElementById("hometown").innerHTML = `${"Baku"}`
> The template-literal pattern is used when you want to compose a text from combination of text and variables, but here you can simply use regular strings
> 
> document.getElementById("nickname").innerHTML = "Nizami Mursudlu"
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/HackYourHomework/JavaScript2/pull/11#pullrequestreview-557375170>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AMK2MGKI76TYKF2FPDNQSBLSWEMLHANCNFSM4UL6XKDQ>.
>

eidosam commented:
> I got a little confused by your last screenshot… Why do I have to write JS if everything is already displayed in HTML…

Sorry if it was not clear, the screenshot is taken from Chrome DevTools, specifically [Elements panel](https://developers.google.com/web/tools/chrome-devtools#elements), which shows the final version of your HTML page after executing JavaScript and CSS. So I was showing you what your JS code producing.
The issue was:
- The request is to produce a single `ul` contains `li` for each book in the list.
- The implemented solution was producing `p` element for each book, and these `p` elements were outside the `ul`.

The new code you pushed has fixed this issue
<img width="772" alt="Nizami-W1-Ex1" src="https://user-images.githubusercontent.com/17527017/103438711-2c4b2800-4c36-11eb-87f1-890e06a82a57.png">

eidosam reviewed (approved):
Thank you, Nizami! Well done!
